packages/app/lib/featureFlags.ts

```diff
@@ -1,33 +1,54 @@
+import { mapValues } from 'lodash';
+import create from 'zustand';
+
 import storage from './storage';
 
+// Add new feature flags here:
 export const featureMeta = {
   channelSwitcher: {
     default: false,
     label: 'Experimental channel switcher',
   },
-};
+} satisfies Record<string, { default: boolean; label: string }>;
 
 export type FeatureName = keyof typeof featureMeta;
 
 export type FeatureState = {
   [K in FeatureName]: boolean;
 };
 
-const featureState: { [K in FeatureName]: boolean } = {
-  channelSwitcher: false,
-};
+interface FeatureStateStore {
+  flags: FeatureState;
+  setEnabled: (name: FeatureName, enabled: boolean) => void;
+}
+
+export const useFeatureFlagStore = create<FeatureStateStore>((set) => ({
+  flags: mapValues(featureMeta, (meta) => meta.default),
```

Comment:
@dnbrwstr "fixed" default types by deferring to `mapValues` – I think in theory this is unsound (e.g. someone could import `featureMeta` and add an entry, which would break a bunch of these types), so w/e, it works lol